test/e2e/space_test.go

```diff
@@ -128,29 +125,118 @@ func TestSpace(t *testing.T) {
+					require.NoError(t, err)
+
+					t.Run("space should be finally deleted", func(t *testing.T) {
+						// when
+						err = hostAwait.WaitUntilSpaceDeleted(s.Name)
+						// then
+						require.NoError(t, err)
+					})
+				})
 			})
 		})
 	})
 }
 
+func TestUpdateSpace(t *testing.T) {
+
+	// given
+
+	// make sure everything is ready before running the actual tests
+	awaitilities := WaitForDeployments(t)
+	hostAwait := awaitilities.Host()
+	memberAwait := awaitilities.Member1()
+
+	rand.Seed(time.Now().UnixNano())
+
+	space := &toolchainv1alpha1.Space{
+		ObjectMeta: v1.ObjectMeta{
+			Namespace: hostAwait.Namespace,
+			Name:      "oddity-" + strconv.Itoa(rand.Int()),
+		},
+		Spec: toolchainv1alpha1.SpaceSpec{
+			TargetCluster: memberAwait.ClusterName,
+			TierName:      "base",
+		},
+	}
+
+	// when
+	err := hostAwait.Client.Create(context.TODO(), space)
+
+	// then
+	require.NoError(t, err)
+	// wait until NSTemplateSet has been created and Space is in `Ready` status
+	space, err = hostAwait.WaitForSpace(space.Name,
+		wait.UntilSpaceHasConditions(Provisioned()),
+		wait.UntilSpaceHasStatusTargetCluster(memberAwait.ClusterName),
+	)
+	require.NoError(t, err)
+	// wait until NSTemplateSet has been created and Space is in `Ready` status
+	nsTmplSet, err := memberAwait.WaitForNSTmplSet(space.Name, wait.UntilNSTemplateSetHasConditions(Provisioned()))
+	require.NoError(t, err)
+	tierChecks, err := tiers.NewChecks("base")
+	require.NoError(t, err)
+	tiers.VerifyGivenNsTemplateSet(t, memberAwait, nsTmplSet, tierChecks, tierChecks, tierChecks.GetExpectedTemplateRefs(hostAwait))
+
+	t.Run("update tier", func(t *testing.T) {
+		// given
+		ctr := NewChangeTierRequest(hostAwait.Namespace, space.Name, "advanced")
+
+		// when
+		err = hostAwait.Client.Create(context.TODO(), ctr)
+
+		// then
+		require.NoError(t, err)
+		err = hostAwait.WaitUntilChangeTierRequestDeleted(ctr.Name)
+		space, err = hostAwait.WaitForSpace(space.Name, wait.UntilSpaceHasTier("advanced"), wait.UntilSpaceHasConditions(Provisioned()))
+		require.NoError(t, err)
+		nsTmplSet, err := memberAwait.WaitForNSTmplSet(space.Name, wait.UntilNSTemplateSetHasConditions(Provisioned()))
+		require.NoError(t, err)
+		tierChecks, err := tiers.NewChecks("advanced")
+		require.NoError(t, err)
+		tiers.VerifyGivenNsTemplateSet(t, memberAwait, nsTmplSet, tierChecks, tierChecks, tierChecks.GetExpectedTemplateRefs(hostAwait))
+		space, err = hostAwait.WaitForSpace(space.Name, wait.UntilSpaceHasConditions(Provisioned()))
+		require.NoError(t, err)
+
+		t.Run("delete space", func(t *testing.T) {
+			// now, delete the Space and expect that the NSTemplateSet will be deleted as well,
+			// along with its associated namespace
+
+			// when
+			err = hostAwait.Client.Delete(context.TODO(), space)
+
+			// then
+			require.NoError(t, err)
+			err = hostAwait.WaitUntilSpaceDeleted(space.Name)
+			require.NoError(t, err)
+		})
+	})
```

Comment:
```suggestion
```
I guess testing deletion isn't really required as part of the update tier test? Unless you think it provides some specific value to the test

Reply:
not really, indeed, but I wanted to clean up the resources at the end of the test. But I'm fine with removing this subtest for the sake of clarity/understanding, and as you said, because it's out of purpose